operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/BulkDataImportUtil.java

```diff
@@ -31,66 +31,79 @@
-                        && parameter.getValue().is(com.ibm.fhir.model.type.String.class)) {
-                    // If the parameter isn't passed, use application/fhir+ndjson
-                    // Check the MediaType
-                    String val = parameter.getValue().as(com.ibm.fhir.model.type.String.class).getValue();
-                    if (BulkDataConstants.INPUT_FORMATS.contains(val)) {
-                        return val;
-                    }
+    /**
+     * processes the parameter inputFormat from the Parameters object and evaluationContext.
+     *
+     * @implNote If there are multiple entries, the processing only takes the first entry that matches.
+     *
+     * @return
+     * @throws FHIROperationException
+     */
+    public String retrieveInputFormat() throws FHIROperationException {
+        try {
+            Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, "parameter.where(name = 'inputFormat').value");
+            Iterator<FHIRPathNode> iter = result.iterator();
+            while (iter.hasNext()) {
+                FHIRPathElementNode node = (FHIRPathElementNode) iter.next();
+                String val = node.asElementNode().element().as(com.ibm.fhir.model.type.String.class).getValue();
+                if (BulkDataConstants.INPUT_FORMATS.contains(val)) {
+                    return val;
                 }
             }
+        } catch (ClassCastException ce) {
+            throw buildExceptionWithIssue("$import invalid parameter name in 'inputFormat'", ce, IssueType.INVALID);
+        } catch (FHIRPathException e) {
+            throw buildExceptionWithIssue("$import invalid parameters value type in 'inputFormat'", e, IssueType.INVALID);
         }
 
         throw buildExceptionWithIssue("$import requires 'inputFormat' is not found", IssueType.INVALID);
     }
 
-    public static String retrieveInputSource(Parameters parameters) throws FHIROperationException {
-        // Parameter: inputSource (required)
-        // If there are multiple entries, the processing only takes the first entry.
-        if (parameters != null) {
-            for (Parameters.Parameter parameter : parameters.getParameter()) {
-                if (BulkDataConstants.PARAM_INPUT_SOURCE.equals(parameter.getName().getValue())
-                        && parameter.getValue() != null && parameter.getValue().is(com.ibm.fhir.model.type.Uri.class)) {
-                    // If the parameter isn't passed, use application/fhir+ndjson
-                    return parameter.getValue().as(com.ibm.fhir.model.type.Uri.class).getValue();
-                }
+    /**
+     * processes the retrieveInputSource from the parameters object and evaluationContext.
+     *
+     * @implNote If there are multiple entries, the processing only takes the first entry.
+     *
+     * @return
+     * @throws FHIROperationException
+     */
+    public String retrieveInputSource() throws FHIROperationException {
```

Comment:
looks pretty similar to the previous method...wonder if it could be a single helper

Reply:
probably not worth it if we only have 2

Reply:
Yeah - I left them separate, but very similar as they are different types.